ex_feels/assets/js/timefeels.js

```diff
@@ -1,393 +1,105 @@
-      .attr('transform', 'rotate(-90)')
-      .attr('y', 6)
-      .attr('dy', '.71em')
-      .style('text-anchor', 'end')
-      .text('Time (s)');
+      .attr('transform', 'translate(-10,' + chartHeight/2 + ')rotate(-90)')
+      .attr('y', 0)
+      .style('text-anchor', 'middle')
+      .text('Sentiment')
 }
 
 function drawPaths (svg, data, x, y) {
-  var upperInnerArea = d3.svg.area()
-    .interpolate('basis')
-    .x (function (d) { return x(d.date) || 1; })
-    .y0(function (d) { return y(d.pct75); })
-    .y1(function (d) { return y(d.pct50); });
 
-  var medianLine = d3.svg.line()
+  var Area = d3.svg.area()
     .interpolate('basis')
-    .x(function (d) { return x(d.date); })
-    .y(function (d) { return y(d.pct50); });
-
-  var lowerInnerArea = d3.svg.area()
+    .x (function (d) { return x(d.time) || 1; })
+    .y0(function (d) { return y(d.mean + d.std**2); }) // use variance instad
+    .y1(function (d) { return y(d.mean - d.std**2); });// of std deviation
+  
+  var meanLine = d3.svg.line()
     .interpolate('basis')
-    .x (function (d) { return x(d.date) || 1; })
-    .y0(function (d) { return y(d.pct50); })
-    .y1(function (d) { return y(d.pct25); });
+    .x(function (d) { return x(d.time); })
+    .y(function (d) { return y(d.mean); });
+
+  // append the gradient 
+  var uGradient = svg.append("defs")
+    .append("linearGradient")
+    .attr("gradientUnits", "objectBoundingBox")
+    .attr("id", "upperGradient")
+    .attr("x1", "0%")
+    .attr("x2", "0%")
+    .attr("y1", 1)
+    .attr("y2", -1)
+    .attr("spreadMethod", "pad");
+
+  uGradient.append("stop")
+    .attr("offset", 0)
+    .attr("stop-color", "#ff3860")
+    .attr("stop-opacity", 0.95);
+
+  uGradient.append("stop")
+    .attr("offset", 0.25)
+    .attr("stop-color", "#a4a4a4")
+    .attr("stop-opacity", 0.85);
+
+  uGradient.append("stop")
+    .attr("offset", 0.5)
+    .attr("stop-color", "#23d160")
+    .attr("stop-opacity", 0.95);  
 
   svg.datum(data);
 
-  svg.append('path')
-    .attr('class', 'area upper inner')
-    .attr('d', upperInnerArea)
+  svg.append('path') // add variance cloud
+    .attr('class', 'area inner')
+    .attr('d', Area)
+    .attr('fill', 'url(#upperGradient)')
     .attr('clip-path', 'url(#rect-clip)');
 
-  svg.append('path')
-    .attr('class', 'area lower inner')
-    .attr('d', lowerInnerArea)
-    .attr('clip-path', 'url(#rect-clip)');
-
-  svg.append('path')
-    .attr('class', 'median-line')
-    .attr('d', medianLine)
+  svg.append('path') // add mean trend
+    .attr('class', 'mean-line')
+    .attr('d', meanLine)
     .attr('clip-path', 'url(#rect-clip)');
 }
 
 function startTransitions (chartWidth, rectClip) {
   rectClip.transition().attr('width', chartWidth);
 }
 
-function makeChart (data) {
-  var svgWidth  = 1024,
-      svgHeight = 450,
-      margin = { top: 20, right: 20, bottom: 40, left: 40 },
+function makeChart (data, name) {
+
+  data.forEach(function(d){ d.time = new Date(1000*d.time) });
```

Comment:
Is this needed? If I remove this line and refresh, the plots look the same.

Reply:
And by look the same, I mean the dates change 😄
The conversion to milliseconds seems required but not the `new Date(...`?
We should be able to do the conversion before the data gets to `timefeels.js`

Reply:
The conversion to a datetime object has to happen somewhere. There's an implicit conversion in the `d3.time` call but it wasn't playing nice with our timestamps. I'll experiment some more. i reckon python should do the milliseconds conversion before the timestamp goes into the db

Reply:
From my experimenting, a millisecond utc timestamp (stats.time * 1000) works with `d3.time` but maybe I can't tell it's messing up the plot. I'm trying to avoid the loop through the data.

Reply:
figured it out... i had been doing the conversion at the wrong spot in python. :scream: